rusqlite_migration/src/builder.rs

```diff
@@ -42,14 +43,11 @@ impl<'u> MigrationsBuilder<'u> {
     ///
     /// Panics if no migration with the `id` provided exists.
     #[must_use]
-    pub fn edit(mut self, id: usize, f: impl Fn(&mut M)) -> Self {
+    pub fn edit(mut self, id: usize, f: impl Fn(M) -> M) -> Self {
         if id < 1 {
             panic!("id cannot be equal to 0");
         }
-        f(self
-            .migrations
-            .get_mut(id - 1)
-            .expect("No migration with the given index"));
+        take(&mut self.migrations[id - 1], f);
```

Comment:
Further to my comment on the `take_mut` dep, we get a user-passed closure here, so it’s unclear that the [AssertUnwindSafe](https://doc.rust-lang.org/stable/std/panic/struct.AssertUnwindSafe.html) annotation that `take_mut` uses is warranted.